_rules/table-header-cells-refer-to-data-cells-d0f69e.md

```diff
@@ -0,0 +1,247 @@
+---
+id: d0f69e
+name: All table header cells have assigned data cells
+rule_type: atomic
+description: |
+  This rule checks that each table header has assigned data cells in a table element.
+accessibility_requirements:
+  wcag20:1.3.1: # Info and Relationships (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+acknowledgements:
+  authors:
+    - Jey Nandakumar
+    - Audrey Maniez
+---
+
+## Applicability
+
+The rule applies to any HTML element that has the [semantic role](#semantic-role) of [rowheader](https://www.w3.org/TR/wai-aria-1.1/#rowheader) or [columnheader](https://www.w3.org/TR/wai-aria-1.1/#columnheader) and is a [descendant](https://dom.spec.whatwg.org/#concept-tree-descendant) in the [flat tree](https://drafts.csswg.org/css-scoping/#flat-tree) of an element that:
+
+- has a [semantic role](#semantic-role) of either [table](https://www.w3.org/TR/wai-aria-1.1/#table) or [grid](https://www.w3.org/TR/wai-aria-1.1/#grid); and
+- is [visible](#visible); and
+- is [included in the accessibility tree](#included-in-the-accessibility-tree)
```

Comment:
I think this rule should ignore tables with out cells, and tables with only empty cells. There is nothing wrong with an unpopulated data table that I can see.

Reply:
I think I agree and I'm afraid that goes a bit deeper.
Essentially, you're saying that
```html
<table>
 <tr><th>Foo</th></tr>
</table>
```
Is not failing 1.3.1 because there is no "structure conveyed through presentation" that needs to be programmatically determinable.
I agree with that.

Going further, if I look at
```html
<table>
 <tr><th>Foo</th><th>Bar</th></tr>
 <tr><td>Hello</td></tr>
</table>
```
Then the second `th` (Bar) has no assigned data cell, but there is no "structure conveyed through presentation" for it either, so I don't think that this example is failing 1.3.1, but it is failing the rule.

Looking a bit into [H43](https://www.w3.org/WAI/WCAG21/Techniques/html/H43), which is cited in the Background, there's nothing in it that says header cells must have assigned data cells.

Which is annoying because that's more or less challenging the very basics of this rule 😢 (actually, that's good with the Assumption suggested by @WilcoFiers).